services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -2585,6 +2569,15 @@ private Pair<Boolean, Version> addVersion(
             }
 
             if (sendStartOfPush) {
+              ByteBuffer compressionDictionaryBuffer = null;
+              if (compressionDictionary != null) {
+                compressionDictionaryBuffer =
+                    ByteBuffer.wrap(EncodingUtils.base64DecodeFromString(compressionDictionary));
+              } else if (store.getCompressionStrategy().equals(CompressionStrategy.ZSTD_WITH_DICT)) {
+                throw new VeniceException(
+                    "compression Dictionary should not be empty if CompressionStrategy is ZSTD_WITH_DICT");
```

Comment:
This will break for empty-push when it doesn't go via the KIF mode - which is still new. Can we keep the controller sending an dummy dictionary logic? Are there any issues?

Reply:
yeah, this code change will require some collaboration between VPJ and Controller because of this.
Maybe we can make it backward compatible in Controller  to check whether the request param contains dict or not and if it doesn't for empty push, the Controller will continue to use this dummy dict, otherwise throw exception.
Once VPJ is fully deployed, we can cleanup this logic from Controller.

Reply:
As sending via controller is introduced for samza batch push, I think this flow is still needed even if we add this logic in vpj as well. So added it back with proper comments.